### PR TITLE
Fixed #24675 --Implemented an option to avoid unnecessary queries to MySQL since MySQL 5.6

### PR DESCRIPTION
**Pull request by ssjunior**
https://code.djangoproject.com/ticket/24675#ticket

Since MySQL 5.6 setting MYSQL_AUTO_IS_NULL is unnecessary. Implementing this eliminates one query in every connection resulting in better performance.

http://dev.mysql.com/doc/refman/5.6/en/server-system-variables.html#sysvar_sql_auto_is_null

System Variable	Name	sql_auto_is_null
Variable Scope	Global, Session
Dynamic Variable	Yes
Permitted Values	Type	boolean
Default	0

### CODE REVIEW

**Comment by timgraham** [comment]
I left a comment on the ticket, did you get notified about it?

**Comment by ssjunior** [comment]
Hi @timgraham, I did not. I will take a look to see if I can find another workaround. I will comment on the ticket.

File: django/db/backends/mysql/base.py [hunk 1]
```diff
@@ -279,12 +279,15 @@ def get_new_connection(self, conn_params):
...
-            # NULL.  Disabling this value brings this aspect of MySQL in line with
-            # SQL standards.
-            cursor.execute('SET SQL_AUTO_IS_NULL = 0')
+        # SQL_AUTO_IS_NULL in MySQL controls whether an AUTO_INCREMENT column
+        # on a recently-inserted row will return when the field is tested for
+        # NULL. Disabling this value brings this aspect of MySQL in line with
+        # SQL standards.
+        # Check the sql_auto_is_null status on the MySQL server and if necessary
+        # set it to False.
+        if self.features.sql_auto_is_null():
```
Comment (timgraham) [inline django/db/backends/mysql/base.py]: chop parenthesis cause it's a property now

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
Also, could you try to add a test as described on the ticket?

**Comment by ssjunior** [comment]
Hi Tim. I will do that. I am  travelling right now and I'll do as soon as I am available.

**Comment by timgraham** [comment]
Friendly reminder in case you have a chance to finish this now.

**Comment by timgraham** [comment]
Closing due to inactivity. Please send a new PR when you have time to update it, thanks!

**Comment by neelay-shah** [comment]
this is still not merged to latest release of django, it has been 9 months. what is the eta on it?

**Comment by timgraham** [comment]
Django 1.10

**Pull request closed without merging**